ModuleNotFoundError when using the tests outside application code layout

- [x] Include a detailed description of the bug or suggestion
       I was following the test layout in [Good Integration Practices](https://docs.pytest.org/en/latest/goodpractices.html) and I couldn't get my tests that were in a similar configuration to run or the simple example I created below.

- [ ] `pip list` of the virtual environment you are using

- [x] pytest and operating system versions
       Ubuntu 16.04.2 LTS (GNU/Linux 4.4.0-77-generic x86_64)
       platform linux -- Python 3.6.1, pytest-3.0.7, py-1.4.33, pluggy-0.4.0
- [x] Minimal example if possible
 ```
\test_pytest
    mypkg/
        __init__.py
        app.py
    tests/
        test_app.py
```
and the files
``` python
# test_app.py
import app

# tried:
# import mypkg.app
# import ..app
# from mypkg import app
```
``` python
# app.py
print('It Works')
def test_pass():
    assert True == True

```
Running  pytest from the `test_pytest` directory will result in an error  similar to this one.

`(py36env) ubuntu@ubuntu-xenial:/vagrant/test_pytest$ pytest`
```
ImportError while importing test module '/vagrant/test_pytest/test/test_app.py'.
Hint: make sure your test modules/packages have valid Python names.
test/test_app.py:1: in <module>
    import app
E   ModuleNotFoundError: No module named 'app'
```


I can add to the `sys` path manually by adding a `__init__.py` to `tests/`
``` python
# test/__init__.py
import sys
sys.path.append('/vagrant/test_pytest/mypkg')
```
but that will complicate testing with Travis CI

--- a/doc/en/goodpractices.rst
+++ b/doc/en/goodpractices.rst
@@ -4,6 +4,27 @@
 Good Integration Practices
 =================================================
 
+Install package with pip
+-------------------------------------------------
+
+For development, we recommend to use virtualenv_ environments and pip_
+for installing your application and any dependencies
+as well as the ``pytest`` package itself. This ensures your code and
+dependencies are isolated from the system Python installation.
+
+First you need to place a `setup.py` file in the root of your package with the following minimum content:
+
+     from setuptools import setup, find_packages
+     
+     setup(name="PACKAGENAME", packages=find_packages())
+
+Where `PACKAGENAME` is the name of your package. You can then install your package in "editable" mode by running from the same directory::
+
+     pip install -e .
+
+which lets you change your source code (both tests and application) and rerun tests at will.
+This is similar to running ``python setup.py develop`` or ``conda develop`` in that it installs
+your package using a symlink to your development code.
 
 .. _`test discovery`:
 .. _`Python test discovery`:
@@ -176,19 +197,6 @@ Note that this layout also works in conjunction with the ``src`` layout mentione
 
 tox
 ------
-
-For development, we recommend to use virtualenv_ environments and pip_
-for installing your application and any dependencies
-as well as the ``pytest`` package itself. This ensures your code and
-dependencies are isolated from the system Python installation.
-
-You can then install your package in "editable" mode::
-
-     pip install -e .
-
-which lets you change your source code (both tests and application) and rerun tests at will.
-This is similar to running ``python setup.py develop`` or ``conda develop`` in that it installs
-your package using a symlink to your development code.
 
 Once you are done with your work and want to make sure that your actual
 package passes all tests you may want to look into `tox`_, the

--- a/doc/en/goodpractices.rst
+++ b/doc/en/goodpractices.rst
@@ -12,13 +12,14 @@ for installing your application and any dependencies
 as well as the ``pytest`` package itself. This ensures your code and
 dependencies are isolated from the system Python installation.
 
-First you need to place a `setup.py` file in the root of your package with the following minimum content:
+First you need to place a ``setup.py`` file in the root of your package with the following minimum content:
 
      from setuptools import setup, find_packages
-     
+
+
      setup(name="PACKAGENAME", packages=find_packages())
 
-Where `PACKAGENAME` is the name of your package. You can then install your package in "editable" mode by running from the same directory::
+Where ``PACKAGENAME`` is the name of your package. You can then install your package in "editable" mode by running from the same directory::
 
      pip install -e .
 

